src/main/java/com/atlassian/db/replica/internal/state/ConnectionState.java

```diff
@@ -53,9 +54,16 @@ public ConnectionState(
             @Override
             public Connection create() throws SQLException {
                 if (connectionProvider.isReplicaAvailable()) {
-                    final Connection replicaConnection = connectionProvider.getReplicaConnection();
-                    parameters.initialize(replicaConnection);
-                    return replicaConnection;
+                    try {
+                        final Connection replicaConnection = connectionProvider.getReplicaConnection();
+                        parameters.initialize(replicaConnection);
+                        return replicaConnection;
+                    } catch (SQLException e) {
+                        return replicaFailureStrategy.onFailure(
```

Comment:
After the change readConnection was pointing to the main database connection only when the replica was disabled. The change may affect:
- connection state.
- isConsistent checks.